File: wagerproof-ios-native/Wagerproof/Features/Chat/WagerBotPendingActionAdapter.swift
import Foundation
import WagerproofModels

/// Adapter that turns a chat widget into a `WagerBotActionPreview`-ready
/// payload. Honeydew uses an equivalent
/// `ChatV3PendingActionAdapter.build(...)` to map between the V3 content
/// block model and the older `AssistantV2ActionPreview` data shape.
///
/// We don't currently have multiple action shapes — every widget is just
/// "open the game sheet" — but this seam lets us add saved-pick / share
/// flows later without touching the bubble view.
enum WagerBotPendingActionAdapter {
    struct Decision {
        let widget: WagerBotChatWidget
        let openSheet: Bool
    }

    static func build(widget: WagerBotChatWidget) -> Decision {
        // Today every widget is a tap-to-open-the-game-sheet
        // affordance. We still wrap it in this adapter so when we later
        // add e.g. "save this pick" / "share with agent" widget kinds
        // the dispatch lives in one place instead of fanning out
        // through the bubble.
        Decision(widget: widget, openSheet: true)
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/chat/WagerBotPendingActionAdapter.kt
package com.wagerproof.app.features.chat

import com.wagerproof.core.models.WagerBotChatWidget

/**
 * Port of iOS `WagerBotPendingActionAdapter.swift`. Adapter seam that maps a
 * chat widget to a display decision. Today every widget is "open the game
 * sheet"; kept as a single dispatch point for future saved-pick / share flows.
 */
object WagerBotPendingActionAdapter {
    data class Decision(val widget: WagerBotChatWidget, val openSheet: Boolean)

    fun build(widget: WagerBotChatWidget): Decision = Decision(widget = widget, openSheet = true)
}
